wp/plugin/lib/equality.ml

```diff
@@ -0,0 +1,387 @@
+open !Core_kernel
+open Bap.Std
+open Graphlib.Std
+open Base__Option.Monad_infix
+
+(* some of these pattern matchings are massive; we don't want fragile warnings *)
+[@@@warning "-4"]
+
+module H = Bap.Std.Graphs.Ir
```

Comment:
Is there a logic to these names? What is H? Maybe `Ir` would be a better choice if it doesn't slash with anything

Reply:
hmm not really. I'll swap it out for something a bit more descriptive (`Ir` if I can).

Reply:
You can, I think module names shadow in OCaml (sometimes you need to be careful).